examples/graphics/shader-graph-chunks-builder.html

```diff
@@ -0,0 +1,134 @@
+                var emissiveMap = asset.resource;
+                app.assets.loadFromUrl("../assets/core_nodes.json", "text", function (err, asset) {
+                    var coreNodeList = asset.resource;
+
+                    //create a shader graph builder
+                    var builder = new pc.ShaderGraphBuilder(coreNodeList);
+
+                    //add input parameters
+                    var timeParam = builder.addParam('float', 'uTime', 0.0);
+                    var emissiveTexParam = builder.addParam('sampler2D', 'uEmissiveMap', emissiveMap);
+                    var emissiveColParam = builder.addParam('vec3', 'uEmmisiveColorParam', new pc.Vec3(0.2, 0.3, 0.3));
+
+                    //create and hook up graph nodes
+                    var scrollNode = builder.addNode('add2', builder.addNode('joinVec2', timeParam, timeParam), builder.addNode('uv0'));
+                    var emissiveSampleNode = builder.addNode('texSample', emissiveTexParam, scrollNode);
+                    var emissiveCalcNode = builder.addNode('mul3', emissiveColParam, {node:emissiveSampleNode, port:'rgb'});
+
+                    //hook up outputs
+                    builder.addOutput({node:emissiveSampleNode, port:'g'}, 'float', 'dAlpha');
+                    builder.addOutput(emissiveCalcNode, 'vec3', 'dEmission');
+
+                    //return built shader graph chunk
+                    var shaderGraphChunk = builder.getShaderGraphChunk();
+
+                    //replace model's materials
+                    var originalMaterials=[];
+                    var shaderGraphMaterials=[];
+                    
+                    model.meshInstances.forEach(function (meshInstance) {
+                        if (!(meshInstance.material instanceof pc.StandardNodeMaterial)) {
+                            var index = originalMaterials.indexOf(meshInstance.material);
+
+                            if (index === -1) {
+                                index = shaderGraphMaterials.length;
+
+                                originalMaterials.push(meshInstance.material);                            
+                                shaderGraphMaterials.push(new pc.StandardNodeMaterial(meshInstance.material, shaderGraphChunk));
+                            }
+
+                            meshInstance.material = shaderGraphMaterials[index];
+                            
+                            meshInstance.material.alphaTest = 0.5;
+                            meshInstance.material.update();
+                        }
+                    });
+
+                    //update time parameter and alpha test every frame
+                    app.on("update", function (dt) {
+                        time += dt;
+                        shaderGraphMaterials.forEach(function (material) {
+                            material.setShaderGraphParameter('uTime', time);
```

Comment:
should we have simpler name for this instead of **setShaderGraphParameter**

Reply:
Any suggestions? This is the equivalent to setParameter - but for the shader graph embedded in the (standard node) material - so the function is setting a shader graph parameter...

Reply:
I wonder if you could maybe even override setParameter and do some quick check to see where the parameter needs to be set?

Reply:
okay - I'll try overriding setParameter

Reply:
Overriding the setParameter function works and is definitely nicer - thanks!